src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/ReflectMethods.java

```diff
@@ -717,9 +717,7 @@ public Value toValue(JCExpression expression, Type targetType) {
             try {
                 pt = targetType;
                 scan(expression);
-                return result != null ?
-                        coerce(result, expression.type, targetType) :
-                        null;
+                return result == null || targetType.hasTag(TypeTag.VOID) ? result : coerce(result, expression.type, targetType);
```

Comment:
I'd suggest to format like this:

```
return (result == null || targetType.hasTag(TypeTag.VOID)) ?
    result : coerce(result, expression.type, targetType);
```

As sometimes humans have issue parsing long conditionals :-)